doc/secret_rotation_general.md

```diff
@@ -0,0 +1,96 @@
+# Secret Rotation
+
+Note, this document explains general rotation of secrets.
+
+The instructions to rotate the CCDB keys specifically are in
+[a separate document](secret_rotation.md).
+
+The audience of this document are
+
+  - Developers working on KubeCF.
+
+  - Operators deploying KubeCF.
+
+# Background
+
+One of the features KubeCF (or rather the CF operator it sits on top of)
+provides is the ability to declare secrets (passwords, certificates, ...)
+and have the system automatically generate something suitably random
+for such on deployment, and distribute the results to the pods using them.
+
+This removes the burden from human operators to come up with lots of
+such just to have all the internal components of KubeCF properly wired
+up for secure communication.
+
+However, even with this operators may wish to change such secrets from
+time to time, or on a schedule. In other words, re-randomize the
+board, and limit the lifetime of any particular secret.
+
+As a note on terminology, this kind of change is called
+__rotating a secret__.
+
+This document describes how this can be done, in the context of KubeCF.
+
+# Finding secrets
+
+Retrieve the list of all secrets maintained by a KubeCF deployment via
+
+    kubectl get quarkssecret -n kubecf
+
+To see the information about a specific secret, for example the NATS password, use
+
+    kubectl get quarkssecret -n kubecf kubecf.var-nats-password -o yaml
+
+Note that each quarkssecret has a corresponding regulare k8s secret it
+controls.
+
+    kubectl get secret -n kubecf
+    kubectl get secret -n kubecf kubecf.var-nats-password -o yaml
+
+# Requesting a rotation for a specific secret
+
+We keep using `kubecf.var-nats-password` as our example secret.
+
+To rotate this secret:
+
+  1. Create yaml file for a ConfigMap of the form:
+
+         ---				   
+         apiVersion: v1			   
+         kind: ConfigMap			   
+         metadata:			   
+           name: rotate-kubecf.var-nats-password
+           labels:			   
+             quarks.cloudfoundry.org/secret-rotation: "true"
+         data:				   
+           secrets: '["kubecf.var-nats-password"]'
+
+     Note, while the name of this ConfigMap can be technically
+     anything (allowed by k8s syntax) I personally prefer to use a
```

Comment:
Instead of making the document personal ("I personally"), prefer using a collective terminology. E.g.
`..., we recommend using...`

Reply:
Addressed with commit 5d2c9c3009c76e029cbf5ce26e164e6bd5cd0980